### PR TITLE
Editing data in tsv files for regeffect tables

### PR DESCRIPTION
**Pull request by UAMS16**
The table data and the tsv data are not in sync. Edited the data to show the correct figures and headers from the table. Created another tsv_renderer for TargetEffectsView because the Regulatory Effects Table has its own specific data in it.
image

### CODE REVIEW

**Review by tncowart — CHANGES_REQUESTED** [review]
Did you talk to Alex about the TSV formats?

File: cegs_portal/search/views/v1/reg_effects.py [hunk 1]
```diff
@@ -4,7 +4,10 @@
 from cegs_portal.search.json_templates.v1.feature_reg_effects import feature_reg_effects
 from cegs_portal.search.json_templates.v1.reg_effect import regulatory_effect
 from cegs_portal.search.models import DNAFeature, RegulatoryEffectObservation
-from cegs_portal.search.tsv_templates.v1.reg_effects import reg_effects as re_tsv
+from cegs_portal.search.tsv_templates.v1.reg_effects import reg_effects as re_data
+from cegs_portal.search.tsv_templates.v1.reg_effects import (
```
[the remark this answers is not in the document]
Comment (UAMS16) [reply cegs_portal/search/views/v1/reg_effects.py:8]: Completed

File: cegs_portal/search/views/v1/reg_effects.py [hunk 2]
```diff
@@ -144,12 +148,17 @@ def get_data(self, options, feature_id) -> Pageable[RegulatoryEffectObservation]
         return reg_effects
 
     def get_tsv(self, request, options, data, feature_id):
-        filename = f"{feature_id}_source_for_regulatory_effect_observations_table_data.tsv"
+        if options is not None and options.get("tsv_format", None) == "bed6":
+            filename = f"{feature_id}_source_for_regulatory_effect_observations_table_data.bed"
+        else:
+            filename = f"{feature_id}_source_for_regulatory_effect_observations_table_data.tsv"
+
```
[the remark this answers is not in the document]
Comment (UAMS16) [reply cegs_portal/search/views/v1/reg_effects.py]: Completed

File: cegs_portal/search/templatetags/custom_helpers.py [hunk 3]
```diff
@@ -14,3 +14,11 @@ def format_float(value):
         return f"{value:.2e}"
     else:
         return f"{value:.6f}"
+
+
+@register.filter
```
[the remark this answers is not in the document]
Comment (UAMS16) [reply cegs_portal/search/templatetags/custom_helpers.py:19]: completed

File: cegs_portal/search/tsv_templates/v1/reg_effects.py [hunk 4]
```diff
@@ -1,20 +1,103 @@
...
+            "Effect Size (log2FC)",
+            "Direction",
+            "Significance",
+            "Experiment",
+            "Target",
+        ]
+    )
+
+    for reo in reos:
+        source = reo.sources.all()[0]
```
[the remark this answers is not in the document]
Comment (UAMS16) [reply cegs_portal/search/tsv_templates/v1/reg_effects.py]: Completed

File: cegs_portal/search/tsv_templates/v1/reg_effects.py [hunk 5]
```diff
@@ -1,20 +1,103 @@
...
+            "Direction",
+            "Significance",
+            "Experiment",
+        ]
     )
 
-    for reo in reg_effects_page:
-        first_source = reo.sources.all()[0]
+    for reo in reos:
+        source = reo.sources.all()[0]
```
[the remark this answers is not in the document]
Comment (UAMS16) [reply cegs_portal/search/tsv_templates/v1/reg_effects.py]: completed

File: cegs_portal/search/helpers/options.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+from django import template
+
+register = template.Library()
+
+
+@register.filter
+def is_bed6(options):
```
Comment (tncowart) [inline cegs_portal/search/helpers/options.py:7]: this doesn't need to be registered as a filter, and I think it could be used in non_targeting_reos.py

**Review by tncowart — CHANGES_REQUESTED** [review]

File: cegs_portal/search/views/v1/reg_effects.py [hunk 7]
```diff
@@ -144,12 +148,17 @@ def get_data(self, options, feature_id) -> Pageable[RegulatoryEffectObservation]
         return reg_effects
 
     def get_tsv(self, request, options, data, feature_id):
-        filename = f"{feature_id}_source_for_regulatory_effect_observations_table_data.tsv"
+        if options is not None and options.get("tsv_format", None) == "bed6":
```
Comment (tncowart) [inline cegs_portal/search/views/v1/reg_effects.py:151]: you can use `is_bed6` here

File: cegs_portal/search/views/v1/reg_effects.py [hunk 8]
```diff
@@ -165,5 +174,9 @@ def get_data(self, options, feature_id) -> Pageable[RegulatoryEffectObservation]
 
     def get_tsv(self, request, options, data, feature_id):
         feature = DNAFeature.objects.get(accession_id=feature_id)
-        filename = f"{feature.name}_targeting_regulatory_effect_observations_table_data.tsv"
+        if options is not None and options.get("tsv_format", None) == "bed6":
```
Comment (tncowart) [inline cegs_portal/search/views/v1/reg_effects.py:177]: you can use `is_bed6` here.

File: cegs_portal/search/tsv_templates/v1/reg_effects.py [hunk 9]
```diff
@@ -1,23 +1,110 @@
...
+                    "0",
+                    if_strand(source.strand),
+                    reo.effect_size,
+                    reo.direction,
+                    format_float(reo.significance),
+                    reo.experiment.name,
+                ]
 
-        tsv_data.append(row)
+            tsv_data.append(row)
```
Comment (tncowart) [inline cegs_portal/search/tsv_templates/v1/reg_effects.py:108]: this needs to be indented one more unit to the left, so it's at the same level as `row = [`

File: cegs_portal/search/tsv_templates/v1/reg_effects.py [hunk 10]
```diff
@@ -1,23 +1,110 @@
...
+    return (
+        f"{source.chrom_name}:{source.location.lower}-{source.location.upper}:{if_strand(source.strand)}:{target.name}"
+    )
+
+
+def bed6_output(reos):
+    tsv_data = []
+    for reo in reos:
+        for target in reo.targets.all():
+            for source in reo.sources.all():
```
Comment (tncowart) [inline cegs_portal/search/tsv_templates/v1/reg_effects.py:15]: If there are no targets this won't add anything to tsv_data.

File: cegs_portal/search/tsv_templates/v1/reg_effects.py [hunk 11]
```diff
@@ -1,23 +1,110 @@
...
-            reo.accession_id,
-            reo.effect_size,
-            reo.direction,
-            reo.significance,
-            first_source.closest_gene_distance,
-            first_source.get_feature_type_display(),
-            first_source.accession_id,
-            reo.experiment.name,
+    for reo in reos:
+        for target in reo.targets.all():
```
Comment (tncowart) [inline cegs_portal/search/tsv_templates/v1/reg_effects.py:51]: If there are no targets this won't return any data.

**Review by tncowart — CHANGES_REQUESTED** [review]

File: cegs_portal/search/tsv_templates/v1/reg_effects.py [hunk 12]
```diff
@@ -1,23 +1,126 @@
-def reg_effects(get_data):
-    reg_effects_page = get_data
+from cegs_portal.search.helpers.options import is_bed6
+from cegs_portal.search.templatetags.custom_helpers import format_float, if_strand
+
+
+def tested_element(source, target):
+    return f"{source.chrom_name}:{source.location.lower}-{source.location.upper}:{if_strand(source.strand)}:{target.name if target else 'N/A'}"
```
Comment (tncowart) [inline cegs_portal/search/tsv_templates/v1/reg_effects.py:6]: The places where you call this function also tests if target is None (`tested_element(source, target) if target else "N/A"`). Which is correct - having the full name or having just "N/A"? My preference is for the full name, but with a small change:

```suggestion
    return f"{source.chrom_name}:{source.location.lower}-{source.location.upper}:{if_strand(source.strand)}{f':{target.name}' if target else ''}"
```

File: cegs_portal/search/tsv_templates/v1/reg_effects.py [hunk 13]
```diff
@@ -1,23 +1,126 @@
...
-            reo.significance,
-            first_source.closest_gene_distance,
-            first_source.get_feature_type_display(),
-            first_source.accession_id,
-            reo.experiment.name,
+    for reo in reos:
+        sources = reo.sources.all()
+        targets = reo.targets.all()
+
+        if not targets:
```
Comment (tncowart) [inline cegs_portal/search/tsv_templates/v1/reg_effects.py:58]: 👍

File: cegs_portal/search/tsv_templates/v1/reg_effects.py [hunk 14]
```diff
@@ -1,23 +1,126 @@
...
+    return f"{source.chrom_name}:{source.location.lower}-{source.location.upper}:{if_strand(source.strand)}:{target.name if target else 'N/A'}"
+
+
+def bed6_output(reos):
+    tsv_data = []
+    for reo in reos:
+        sources = reo.sources.all()
+        targets = reo.targets.all()
+
+        if not targets:
```
Comment (tncowart) [inline cegs_portal/search/tsv_templates/v1/reg_effects.py:15]: 👍

**Review by tncowart — CHANGES_REQUESTED** [review]

File: cegs_portal/search/tsv_templates/v1/reg_effects.py (hunk 12 above)
Comment (tncowart) [reply cegs_portal/search/tsv_templates/v1/reg_effects.py]: per Alex, leave the colon.

**Review by tncowart — APPROVED** [review]
